frontend/src/metabase/dashboard/components/DashCard/DashCardVisualization.tsx

```diff
@@ -193,24 +222,60 @@ function DashCardVisualization({
           isResultDirty: false,
         }));
 
+    const handleModalOpen = async () => {
+      try {
+        if (!chartPublicUuid) {
+          const public_uuid = await getChartUuid(dashcard.card.id);
+          setChartPublicUuid(public_uuid);
+        }
+        setModalOpen(true);
+      } catch (error) {
+        console.error("Error fetching chart API data:", error);
+      }
+    };
+
+    const handleModalClose = () => {
```

Comment:
instead of handleModalOpen and handleModalClose
You can manage in only one function `handleModal` 

By this condition
`setModal(!modal)`
Instead of setting true or false 
This will work for both open and close of modal